pkg/broker/file.go

```diff
@@ -101,20 +101,17 @@ func newDataFrom(kubeClient kubernetes.Interface, brokerNamespace, ipsecFile str
 		return nil, errors.Wrap(err, "error getting broker client secret")
 	}
 
-	if ipsecFile != "" {
-		ipsecData, err := ReadInfoFromFile(ipsecFile)
+	psk := ipsecPSK
+	if len(psk) == 0 {
+		psk, err = generateRandomPSK(ipsecSecretLength)
 		if err != nil {
-			return nil, errors.Wrapf(err, "error importing IPsec PSK from file %q", ipsecFile)
+			return nil, err
 		}
```

Comment:
I think this logic needs to be outside the scope of `newDataFrom` and `WriteInfoToFile`, it should be part of `deployBrokerInContext` right after trying to read data from file.

E.G. `WriteInfoToFile` should do what it says and not decide if missing info should be added or not, that decision should happen when calling it (the recover command shouldn't invent a PSK if it's missing)